crates/blockless-drivers/src/http_driver/reqwest_driver.rs

```diff
@@ -58,22 +66,35 @@ pub(crate) async fn http_req(url: &str, opts: &str) -> Result<(u32, i32), HttpEr
     let connect_timeout = json["connectTimeout"]
         .as_u64()
         .map(|s| Duration::from_secs(s));
-    let read_timeout = json["readTimeout"].as_u64().map(|s| Duration::from_secs(s));
+    let read_timeout = json["readTimeout"]
+        .as_u64()
+        .map(|s| Duration::from_secs(s));
 
-    // build the headers from the options json
+    // build the headers from the options json  
     let mut headers = reqwest::header::HeaderMap::new();
     let header_value = &json["headers"];
-    let header_obj = match json::parse(header_value.as_str().unwrap()) {
+    
+    // Check if header_value is a valid string
+    let header_obj = match json::parse(header_value.as_str().unwrap_or_default()) {
         Ok(o) => o,
         Err(_) => return Err(HttpErrorKind::HeadersValidationError),
     };
 
     if header_obj.is_object() {
         for (key, value) in header_obj.entries() {
-            headers.insert(
-                reqwest::header::HeaderName::from_bytes(key.as_bytes()).unwrap(),
-                reqwest::header::HeaderValue::from_str(value.as_str().unwrap()).unwrap(),
-            );
+            // Handle possible errors from from_bytes
+            let header_name = match reqwest::header::HeaderName::from_bytes(key.as_bytes()) {
+                Ok(name) => name,
+                Err(_) => return Err(HttpErrorKind::HeadersValidationError),
+            };
+
+            // Handle possible errors from from_str
+            let header_value = match reqwest::header::HeaderValue::from_str(value.as_str().unwrap_or_default()) {
+                Ok(value) => value,
+                Err(_) => return Err(HttpErrorKind::HeadersValidationError),
+            };
```

Comment:
nit: simplification:
```suggestion
        let header_name = reqwest::header::HeaderName::from_bytes(key.as_bytes())
            .map_err(|_| HttpErrorKind::HeadersValidationError)?;

        let header_value = reqwest::header::HeaderValue::from_str(value.as_str().unwrap_or_default())
            .map_err(|_| HttpErrorKind::HeadersValidationError)?;
```

Reply:
Thanks! I like this! I clean it up!